pkg/shardddl/optimism/lock_test.go

```diff
@@ -1522,7 +1657,9 @@ func (t *testLock) TestAddDifferentFieldLenColumns(c *C) {
 
 	// TrySync for the second table, add a table with a smaller field length
 	vers[source][db][tbls[0]]++
-	DDLs, err = l.TrySync(source, db, tbls[0], DDLs1, []*model.TableInfo{ti1}, tts, vers[source][db][tbls[0]])
+	info = NewInfo(task, source, db, tbls[0], downSchema, downTable, DDLs1, nil, []*model.TableInfo{ti1})
+	info.Version = vers[source][db][tbls[0]]
```

Comment:
You can add a function `NewInfoWithVersion` in this `test.go` since it's used so many times.